@xen-orchestra/lite/src/views/story/HomeView.vue

```diff
@@ -0,0 +1,197 @@
+<template>
+  <div class="home-view">
+    <UiTitle type="h4">
+      This helper will generate a basic story component
+    </UiTitle>
+    <div>
+      Choose a component :
+      <select v-model="componentPath">
+        <option value="" />
+        <option v-for="(component, path) in componentsWithProps" :key="path">
+          {{ path }}
+        </option>
+      </select>
+      <div class="slots">
+        <label>
+          Slots names, separated by a comma
+          <input v-model="slots" />
+        </label>
+        <button @click="slots = 'default'">Default</button>
+        <button @click="slots = ''">Clear</button>
+      </div>
+    </div>
+    <CodeHighlight v-if="componentPath" :code="template" />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import CodeHighlight from "@/components/CodeHighlight.vue";
+import UiTitle from "@/components/ui/UiTitle.vue";
+import { type ComponentOptions, computed, ref, watch } from "vue";
+
+const componentPath = ref("");
+
+const components = import.meta.glob<{ default: ComponentOptions }>(
+  "/src/components/**/*.vue",
+  {
+    eager: true,
+  }
+);
+
+const componentsWithProps = Object.fromEntries(
+  Object.entries(components).filter(
+    ([, component]) => component.default.props || component.default.emits
+  )
+);
+
+const lines = ref<string[]>([]);
+const slots = ref("");
+
+const quote = (str: string) => `'${str}'`;
+const paramsToImport = ref(new Set<string>());
+const widgetsToImport = ref(new Set<string>());
+
+const template = computed(() => {
+  const componentName = componentPath.value.replace(/.*\/([^/]+)\.vue$/, "$1");
+  const path = componentPath.value.replace(/^\/src/, "@");
+  const paramsLines = [...lines.value];
+  const slotsNames = slots.value
+    .split(",")
+    .map((s) => s.trim())
+    .filter(Boolean);
```

Comment:
[`.filter(Boolean)`](https://medium.com/coding-at-dawn/kent-c-dodds-filter-trick-will-change-how-you-use-javascript-87b5112f9f6d) is a [really common way](https://github.com/search?l=JavaScript&p=99&q=%22.filter%28Boolean%29%22&type=Code) to filter out falsy values from an array :)

I'm OK to do the change here, because we know the array contains strings only, but don't be surprised to see this "trick" later ^^